### ISSUE TITLE
F1 score metric

### ISSUE THREAD

**Issue by aiqc** [tensorflow/tensorflow#56358]
### Issue Type

Feature Request

### Source

source

### Tensorflow Version

2.9.1

### Custom Code

No

### Current Behaviour?

```shell
No F1 score metric. It's a holistic measure for classification.

https://torchmetrics.readthedocs.io/en/stable/classification/f1_score.html

https://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html#sklearn.metrics.f1_score
```

### Standalone code to reproduce the issue

```shell
https://www.tensorflow.org/api_docs/python/tf/keras/metrics
```

**Comment by bhack** [comment]
Check https://github.com/keras-team/keras-cv/issues/393

**Comment by sachinprasadhs** [comment]
@aiqc, You can make use of the metrics class from keras-metrics for F1 score, [here](https://github.com/netrack/keras-metrics/blob/master/keras_metrics/metrics.py#L228) is the link of the code.

**Comment by aiqc** [comment]
@sachinprasadhs ah, it is not in the metrics documentation. 
https://www.tensorflow.org/api_docs/python/tf/keras/metrics

**Comment by sachinprasadhs** [comment]
@aiqc , In document only those classes are mentioned which are exported.
Below is the example.
https://github.com/keras-team/keras/blob/8024bd3bd6da31adcbcfda28dacfc66f74e1fb20/keras/metrics/metrics.py#L1605-L1606

**Comment by bhack** [comment]
@sachinprasadhs These metrics are third_party/deprecated.

You could use TF Addons if you don't want to wait Keras-cv porting.

**Comment by sachinprasadhs** [comment]
@bhack ,Thanks for the suggestion. 
@aiqc , Could you please make use of the metric from TF Addons from the below link. 

> @sachinprasadhs These metrics are third_party/deprecated.
> 
> You could use TF Addons if you don't want to wait Keras-cv porting:
> 
> https://www.tensorflow.org/addons/api_docs/python/tfa/metrics/F1Score

**Comment by aiqc** [comment]
This is a feature request to get F1 in core metrics for feature parity with other ML tools

**Comment by sachinprasadhs** [comment]
You can create this feature request in keras repo since Development of keras moved to separate repository https://github.com/keras-team/keras/issues

To know more see;
https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999
Thank you!
